src/CommunityToolkit/Xamarin.CommunityToolkit/Views/BadgeView/BadgeView.shared.cs

```diff
@@ -328,12 +328,13 @@ async Task UpdateVisibilityAsync()
 
 			if (string.IsNullOrEmpty(badgeText))
 			{
-				IsVisible = false;
+				BadgeIndicatorBackground.IsVisible = false;
 				return;
 			}
 
 			var badgeIsVisible = !AutoHide || !badgeText.Trim().Equals("0");
-
+			BadgeIndicatorBackground.IsVisible = badgeIsVisible;
+			
```

Comment:
Seems this line is redundant (We set badgeIsVisible below)

Reply:
we set for `BadgeIndicatorContainer` not for `BadgeIndicatorBackground` in line 357. If this line removed together with 331 then a red frame will be visible.